packages/engine/src/worker/runner/python/sender.rs

```diff
@@ -118,6 +118,8 @@ impl NngSender {
 
 impl Drop for NngSender {
     fn drop(&mut self) {
+        // Python has to close the socket on its side for the "-topy*" file to be deleted.
+        let _ = self.send(None, &InboundToRunnerMsgPayload::TerminateRunner, &None);
```

Comment:
I don't think this is the right location to send this to Python. Also, we have methods on the worker level to end a worker process. In case an experiment completes, this function is called from there as well, we should use the same behavior and wait for the process to end (there is even a TODO for this exact thing in the Python runner right before dropping NngSender).

Reply:
The todo thing wouldn't be called if the engine panics.
I agree it'll try to send a message even if python is done, in that case I don't it'll do much.

Reply:
In this case, we always send a termination message twice. `Nng` should really only have knowledge about its socket, not about any engine-specific logic.
Then, it would be better to wrap the python process and wait for its completion when dropping or even using `#[panic_handler]`. Anyway, we can't guarantee, that the Python runner will clean up the files because it may already have crashed, so generally this does not solve our problem: Clean up an experiment if bad things happened.

Reply:
As you re-requested a review: What is the status on this?

Reply:
While I like the idea of having a `PythonProcess` struct, the issue still remains: Why we need to send it here in the first place? The engine will send this command on a gracefully finished experiment. My previous comment:

> Anyway, we can't guarantee, that the Python runner will clean up the files because it may already have crashed, so generally this does not solve our problem: Clean up an experiment if bad things happened.

Reply:
The goal of sending it on `Drop` is to be able to send it even when the engine panics in a random place.
And yes the python runner might have already crashed, in that case too bad, sending a message won't do anything.
In basically any other cases it'll allow the python runner to properly cleanup on its own.

When the python runner crashes then the `orchestrator` (or something similar) will be there to cleanup.

Reply:
Calling it on Drop can't guarantee, that the python process wasn't dropped before (there is no specified Drop order in Rust). If the orchestrator will clean it up on a crash, why do we even need this?